### DISCUSSION TITLE
Use a reverse proxy to expose only the necessary CA endpoints. Caddy.

### DISCUSSION THREAD

**Discussion by lukasz-lobocki** [smallstep/certificates#1641] [Q&A] [answered]
I am trying to make my ca available from public internet. I successfully did it by forwarding the whole port `:9000` on my router (my step-ca operates on this port).

However I would like to follow this [advice](https://smallstep.com/docs/step-ca/certificate-authority-server-production/#exposing-step-ca-to-the-internet), but I do not know how to do it with Caddy (note: my caddy and my smallstep are on 2 different internal hosts).

I tried this in Caddyfile:
```
https://subdomain.domain.duckdns.org {
  reverse_proxy https://step-ca_hostname:9000 {
    transport http {
      tls_insecure_skip_verify
    }
  }
}
```
but when issuing on client
```
step ca certificate --ca-url=https://subdomain.domain.duckdns.org lukasz tmp3.crt tmp3.key
```
I am getting
```
client GET https://subdomain.domain.duckdns.org/provisioners?limit=100 failed: x509: certificate signed by unknown authority
```

**Comment by tashian** [comment, marked as answer]
Hi! Good question.

Check out the section [Exposing your CA to the internet](https://smallstep.com/blog/access-your-homelab-anywhere/#exposing-your-ca-to-the-internet) in our blog post [Access your homelab from anywhere with a YubiKey and mutual TLS](https://smallstep.com/blog/access-your-homelab-anywhere/). It shows how to proxy the ACME endpoints, but you could expose more (as described in the docs that you mentioned) if you want.

Hope this helps.
